templates/terraform/main.tf

```diff
@@ -0,0 +1,81 @@
+provider "aws" {
+  region              = "${var.region}"
+}
+
+# {{ if .Config.Infrastructure.AWS.Terraform.RemoteState }}
+# Store remote state in S3
+resource "aws_s3_bucket" "terraform_remote_state" {
+  bucket  = "${ var.remote_state_s3_bucket }"
+  acl     = "private"
+
+  versioning {
+    enabled = true
+  }
+}
+
+resource "aws_dynamodb_table" "terraform_state_locks" {
+  name           = "${ var.remote_state_dynamo_table }"
+  read_capacity  = 2
+  write_capacity = 2
+  hash_key       = "LockID"
+
+  attribute {
+    name = "LockID"
+    type = "S"
+  }
+}
+
+# Reference the remote state
+terraform {
+  backend "s3" {
+    bucket         = "${var.remote_state_s3_bucket}"
+    key            = "infrastructure/terraform/shared"
+    encrypt        = true
+    region         = "${var.region}"
+    dynamodb_table = "${var.remote_state_dynamo_table}"
+  }
+}
+# {{- end}}
+# {{ if .Config.Infrastructure.AWS.Cognito.Deploy }}
+resource "aws_cognito_user_pool" "users" {
```

Comment:
All this stuff already exists in the environment module, why is it duplicated again in main?

Reply:
Or if it's not there already it should be there.